lib/camp.js

```diff
@@ -629,13 +629,16 @@ function respondWithFile(req, res, path, ifNoFile) {
     }
     res.mime(p.extname(realpath).slice(1));
 
-    // Cache management.
-    if (+(new Date(req.headers['if-modified-since'])) >= +stats.mtime) {
-      res.statusCode = 304;
-      res.end();  // not modified.
+    // Cache management (compare timestamps at second-level precision).
+    var modified = new Date(stats.mtime);
+    modified.setMilliseconds(0);
```

Comment:
I would recommend performing integer operations rather than Date operations, given the tremendous number of caveats JS Dates have as a consequence of their Java inheritance.